core/serialisation/src/test/java/uk/gov/gchq/gaffer/serialisation/implementation/StringSerialiserTest.java

```diff
@@ -55,6 +60,15 @@ public void canSerialiseStringClass() {
         assertTrue(serialiser.canHandle(String.class));
     }
 
+    @Test
+    public void shouldSerialiseEmpty() throws SerialisationException {
+        // When
+        final byte[] bytes = serialiser.serialise("");
+
+        // Then
+        assertArrayEquals(EMPTY_BYTES, bytes);
```

Comment:
can you have little look to see if other things might be getting empty concats too.

`,,,,,`

Reply:
@GCHQDev404  So I did some digging in the serialisers that implement `ToBytesSerialiser` and found some interesting results. For most serialisers, like Long, Float and Boolean, this problem doesn't apply as those data types cannot be initialised empty.

But for quite a few others there is indeed an issue with serialising empty values and having them deserialised as `null`, like what used to happen with String before this PR. For example, if you serialise `new HashMap()` with the `MapSerialiser` it will be deserialised as `null`, not an empty map.

Some serialisers behave the opposite way too. For example, using a `SetSerialiser`, when `null` is serialised and then deserialsed the result is an empty `Set`.

However, these inconsistencies are not quite as bad as the old `StringSerialiser`, as `StringConcat` only applies to Strings. For other `BinaryOperators` that aggregate properties, such as `First`, this doesn't produce quite such a strange result, but it is still inconsistent.

Either way, I think this discussion on whether we should make all of the serialisers consistent is worthy of another issue and out of the scope of this PR which just changes the `StringSerialiser`. It might be the case though that we do choose to make all of the serialisers consistent and wait to merge this PR until then, perhaps in Gaffer v2.

Reply:
The reason why we had `null` serialsiation is that storing "a lot" of empty maps takes up "a lot" of data.

Reply:
That makes sense, I would be okay with leaving it like that but there are a few serialisers, like Set, that do the opposite. If you give it `null` it actually serialises to an empty Set.